### DISCUSSION TITLE
HOLD for drones

### DISCUSSION THREAD

**Discussion by OrxataEnFartons** [TUDelft-CNS-ATM/bluesky#425] [Ideas]
Sorry for abusing the forum but if I am going to find something to comment or suggest it will be these days that I am working with this. 
Just a suggestion.
I would like to send instructions to the drones to hold, or wait. Concretely, I could receive also a number representing the number of seconds to keep on hold, or a -1 to keep holding until a _continue_ arrives. 
I see there isn't a command for that (HOLD stops the whole simulation), I guess that because the simulator was not addressed to drones originally. So I thought: what if I send a **SPD acid 0**? I tried, I don't get any error but the drone keeps advancing at least at: CAS/TAS/GS: 126/159/159 kts M: 0.253

I can think of an evil way to get it, that is starting a loop with a MOVE constantly resetting it in its current position during the requested holding time. But it would be nice to have the possibility to cleanly stop it with the SPD, 

Regards.

**Comment by ProfHoekstra** [comment]
Dear Miquel,

Only limits to speed are due to the performance model, but I would think the rotor models do allow for speed 0.

Best regards,
    Jacco Hoekstra


On 20 Sep 2022, at 16:46, Miquel Angel ***@***.***> wrote:

﻿

Sorry for abusing the forum but if I am going to find something to comment or suggest it will be these days that I am working with this.
Just a suggestion.
I would like to send instructions to the drones to hold, or wait. Concretely, I could receive also a number representing the number of seconds to keep on hold, or a -1 to keep holding until a continue arrives.
I see there isn't a command for that (HOLD stops the whole simulation), I guess that because the simulator was not addressed to drones originally. So I though: what if I send a SPD acid 0? I tried, I don't get any error but the drone keeps advancing at least at: CAS/TAS/GS: 126/159/159 kts M: 0.253

I can think of an evil way to get it, that is starting a loop with a MOVE constantly resetting it in its current position during the requested holding time. But it would be nice to have the possibility to cleanly stop it with the SPD,

Regards.

**Comment by jooste** [comment]
Hi @OrxataEnFartons,

If I start out with the example scenario you've shown in an earlier discussion and add the following:
```
00:02:00.00>M6001 SPD 0
00:02:40.00>M6001 SPD 38.8;M6001 LNAV ON
```
The drone correctly slows down to 0, and speeds up again forty seconds later. Could you give a scenario example where this doesn't work?

**Reply by OrxataEnFartons** [reply]
Hi all, 
Soooorry! I sent an edit of the post yesterday a while after publishing it asking you to delete it because I realized I was using a B737 and not a drone model. I found no option to delete my post. So Jacco is right, if it is a multi-rotor model it stops completely. 

About keeping it in hold for a while, the plugin I am working on is to connect Bluesky with a UTM server. During the execution of a scenario, the UTM could send separation instructions to the drone, so I don't know the instructions beforehand to add them to the scn. 
Then, to restore the speed after a holding time I started using a threading.Timer and that worked well; but I just realized we have the DELAY command which is perfect for that:

`restore_cmd = "SPD " + traf.id[idx] + " " + str(traf.gs[idx] * 1.94384)`
`stack.stack("DELAY " + str(cmd["wait"]) + " " + restore_cmd) #cmd["wait"] contains the holding time`

Thanks!

**Comment by jooste** [comment]
Indeed! `DELAY` is much more suited, especially since threading.Timer relates to real (wall) time, whereas `DELAY` is simulation time. If you ever want to run your simulation in fast time it will still work as expected with DELAY.

**Reply by OrxataEnFartons** [reply]
Good point! Never thought of that. Thanks!
